docs/xks/developer-guide/observability.md

```diff
@@ -0,0 +1,244 @@
+---
+id: observability
+title: Observability
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+First what is observability?
+Within observability we normally talk about three pillars.
+
+- metrics
+- logging
+- tracing
+
+Some people could argue that there is more steps to it but for now we will focus on these three.
+
+We currently support two solutions to gather observability data in XKF, [Datadog](https://www.datadoghq.com) and the [Opentelemetry](https://opentelemetry.io/) stack which is a open-source solution.
+
+## Datadog
+
+When Datadog monitoring is used the [Datadog Operator](https://github.com/DataDog/datadog-operator) will be added to the cluster. On top of deploying a
+Datadog agent to every node to collect metrics, logs, and traces it also adds the ability to create Datadog monitors from the cluster. The Datadog
+agent handles all communication with the Datadog API meaning that individual applications do not have to deal with thing such as authentication.
+
+### Logging
+
+All logs written to `stdout` and `stderr` from applications in the tenant namespace will be collected by the Datadog agents. This means that no additional
```

Comment:
You don't have to change any source code, but you still have to configure the k8s deployment right?

Reply:
As long as you send data to stdout and stdrerr is picked up by containerd and stored in the logs of the node. So datadog just grabs all that data. So you don't have to do any configuration in your deployment when it comes to telling datadog where to find the logs.

I had this infomration at the grafana-agent section but I decided to remove it since you as a developer don't have to know about this.